docs/EN_US/ECLProgrammersGuide/PRG_Mods/PrGit_Source_Code.xml

```diff
@@ -0,0 +1,610 @@
+      used.</para>
+
+      <para>There are plans to add support for dependencies between multiple
+      repositories in a future release.</para>
+    </sect3>
+  </sect2>
+
+  <sect2>
+    <title>Using Git with HPCC</title>
+
+    <para>The --main syntax has been extended to allow compiling directly from
+    the repository.</para>
+
+    <para>Consider the following command :</para>
+
+    <programlisting>ecl run thor --main demo.main@https://github.com/gituser/gch-demo-d#version1 --server=... </programlisting>
+
+    <para>This command submits a query to Thor via ESP. It retrieves ECL code
+    from the 'version1' branch in the https://hithub.com/gituser/gch-demo-d
+    repository. Compiles the code in the demo/main.ecl file and then runs the
+    query on Thor. The checkout will be done on the remote ECLCC Server rather
+    than on the client machine.</para>
+
+    <sect3 id="RepoReferenceSyntax">
+      <title>Repository Reference Syntax</title>
+
+      <para>The syntax for the reference to the repository is as
+      follows:</para>
+
+      <programlisting>&lt;protocol:&gt;//&lt;urn&gt;/&lt;user&gt;/&lt;repository&gt;#version</programlisting>
+
+      <para>The protocol and urn can be omitted and a default will be used.
+      Such as in the following example:</para>
+
+      <programlisting>ecl run thor --main demo.main@gituser/gch-ecldemo-d#version1 --server=...</programlisting>
+
+      <para>This command also submits a query to Thor, retrieves ECL code from
+      the 'version1' branch in the gch-demo-d repository. Compiles the code in
+      the demo.main.ecl file and then runs the query on Thor.</para>
+    </sect3>
+
+    <sect3 id="HashVersionText">
+      <title>Version-Text</title>
+
+      <para>The version text that follows the hash (#) in the repository
+      reference can take any of the following forms:</para>
+
+      <itemizedlist>
+        <listitem>
+          <para>The name of a branch</para>
+        </listitem>
+
+        <listitem>
+          <para>The name of a tag</para>
+        </listitem>
+
+        <listitem>
+          <para>A SHA of a commit</para>
+        </listitem>
+      </itemizedlist>
+
+      <para>To illustrate consider the following commands:</para>
+
+      <programlisting>ecl run thor --main demo.main@gituser/gch-ecldemo-d#version1 --server=...</programlisting>
+
+      <para>This command will retrieve the demo.main ECL code from the
+      'version1' branch of the gch-ecldemo-d repository.</para>
+
+      <programlisting>ecl run thor --main demo.main@gituser/gch-ecldemo-d#3c23ca0 --server=...</programlisting>
+
+      <para>This command will retrieve the demo.main ECL code from the commit
+      with the secure hash algorithm (SHA) of '3c23ca0'.</para>
+
+      <para><variablelist>
+          <varlistentry>
+            <term>NOTE:</term>
+
+            <listitem>
+              <para>You only need to reference the first 6 characters of a
```

Comment:
This isn't correct.  The number of characters that you need to reference depends on how many commits there are to the repository.  The number is enough to make sure it is unique.  For instance on the HPCC source code repository it is currently 10.